blocks/library/gallery/index.js

```diff
@@ -61,16 +70,11 @@ registerBlockType( 'core/gallery', {
 			query( 'div.blocks-gallery figure.blocks-gallery-image img', {
 				url: attr( 'src' ),
 				alt: attr( 'alt' ),
-			} ),
+			} ) || [],
```

Comment:
I really think most of the problem is a documentation problem. What does each property of the block settings mean. when it's executed. If we take a step back and take a look at the API, we can split it into three different parts:

**1- Parsing attributes from content:**

Currently done using the `attributes` and `defaultAttributes` setting. There are propositions to use a `schema` or `getInitialAttributes` method.

Maybe for clarity, we should simplify this to a `parse( commentAttributes, rawContent ) => 
 blockAttributes` where `rawContent` could be a string (like now) or a tree structure (like proposed elsewhere).

**2- Editing the attributes:**

Once parsed, the block get rendered using the `edit` Component receiving `attributes` and `setAttributes`. I like how this works, I think it's clear enough, maybe I'd just try to find a better name for this settings: `form` or `formComponent`. I don't think we should hide the fact that it's a component like we're trying to do now. 

**3- Serializing attributes to content:**

We do this now using the `save` method.  The `schema` proposed could serve this purpose too, but we're not sure about it yet.

My opinion is that we should make it clear that it's a serialization and that this functions is the opposite of the `parse` function. `serialize( attributes ) => { commentAttributes, rawContent }`. Should `rawContent` here be a string or a tree structure or a React Component. Good question.

---
What I want to say is that we should break our documentation to three steps like this, I think, this makes the API clearer no matter the details.